shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-core/src/main/java/org/apache/shardingsphere/data/pipeline/scenario/rulealtered/RuleAlteredJobPreparer.java

```diff
@@ -91,29 +91,30 @@ private void prepareTarget(final JobConfiguration jobConfig, final PipelineDataS
             log.info("dataSourcePreparer null, ignore prepare target");
             return;
         }
-        LockContext lockContext = PipelineContext.getContextManager().getInstanceContext().getLockContext();
-        ShardingSphereLock lock = lockContext.getOrCreateSchemaLock(jobConfig.getWorkflowConfig().getSchemaName());
+        final PipelineSimpleLock lock = PipelineSimpleLock.getInstance();
         // TODO make sure only the first thread execute prepare
-        boolean skipPrepare = !lock.tryLock("prepareTargetTablesLock", 100);
+        boolean skipPrepare = !lock.tryLock(PREPARE_LOCK_KEY + jobConfig.getHandleConfig().getJobId(), 100);
         if (skipPrepare) {
-            // TODO polling until first thread finish prepare, then just return
-            while (lock.isLocked("prepareTargetTablesLock")) {
+            while (true) {
                 try {
                     TimeUnit.SECONDS.sleep(1);
                 } catch (InterruptedException e) {
-                    lock.releaseLock("prepareTargetTablesLock");
+                    lock.releaseLock(PREPARE_LOCK_KEY + jobConfig.getHandleConfig().getJobId());
+                }
+                // TODO just return ,because the first thread finish prepared
+                if (lock.tryLock(PREPARE_LOCK_KEY + jobConfig.getHandleConfig().getJobId(), 100)) {
+                    lock.releaseLock(PREPARE_LOCK_KEY + jobConfig.getHandleConfig().getJobId());
+                    return;
                 }
             }
-            lock.releaseLock("prepareTargetTablesLock");
-            return;
         }
         try {
             JobDataNodeLine tablesFirstDataNodes = JobDataNodeLine.unmarshal(jobConfig.getHandleConfig().getTablesFirstDataNodes());
             PrepareTargetTablesParameter prepareTargetTablesParameter = new PrepareTargetTablesParameter(tablesFirstDataNodes,
                     jobConfig.getPipelineConfig(), dataSourceManager);
             dataSourcePreparer.prepareTargetTables(prepareTargetTablesParameter);
         } finally {
-            lock.releaseLock("prepareTargetTablesLock");
+            lock.releaseLock(PREPARE_LOCK_KEY + jobConfig.getHandleConfig().getJobId());
         }
```

Comment:
It's better to extract a new method to get lock name, but not hard-coded several times.